src/readers/waffle/waffle.js

```diff
@@ -154,7 +102,80 @@ var WSReader = Reader.extend({
    */
   getData: function () {
     return this._data;
+  },
+
+  _readCallbackSuccess: function (p, path, query, resp) {
+
+    if (!resp) {
+      utils.error("Empty json: " + path);
+      p.reject({
+        'message' : this.ERROR_RESPONSE,
+        'data': path
+      });
+      return;
+    }
+
+    //format data
+    resp = utils.mapRows(this._uzip(resp.data || resp), this._parsers);
+
+    //cache and resolve
+    FILE_CACHED[path] = resp;
+
+    this._parse(p, query, resp);
+    FILE_REQUESTED[path] = void 0;
+  },
+
+  _readCallbackError: function (p, path, query, resp) {
+    p.reject({
+      'message' : this.ERROR_NETWORK,
+      'data': path
+    });
+  },
+
+  _uzip: function (table) {
+    var rows = table.rows;
+    var headers = table.headers;
+    var result = new Array(rows.length);
+    // unwrap compact data into json collection
+    for (var i = 0; i < rows.length; i++) {
+      result[i] = {};
+      for (var j = 0; j < headers.length; j++) {
+        if (typeof rows[i][j] == 'undefined' || rows[i][j] === null) {
+          result[i][headers[j]] = '';
+        } else {
+          result[i][headers[j]] = rows[i][j].toString();
+        }
+        if (headers[j] === 'geo.cat') {
+          result[i][headers[j]] = [result[i][headers[j]]];
+        }
+      }
+    }
+    return result;
+  },
+
+  _parse: function (p, query, resp) {
+
+    var data = resp;
+
+    // sorting
+    // one column, one direction (ascending) for now
+    if(query.orderBy && data[0]) {
```

Comment:
Statement could be restructured by no using `else`. It will increase readability